### ISSUE TITLE
Loading previous chat history

### ISSUE THREAD

**Issue by Seblink** [siilike/skypechatsexporter#1]
Thank you for this wonderful project. It works pretty well.

One thing I encountered is that this only exports chat history that has been loaded before in Skype 8. Normally this isn't a problem, but there are some conversations for me that are only stored in the cloud. When I attempt to scroll back to the start to load them all into the IndexedDB, Skype becomes sluggish and freezes after a few months. I think this is because it hits some kind of Electron memory limit of 2GB.

Since you seem familiar with Electron, I was wondering if you have any solutions such as:
- Increasing the working memory limit of Skype somehow
- Fetching and storing the cloud history into the IndexedDB without having to scroll back in the conversation, perhaps by calling some function that is exposed?

**Comment by siilike** [comment]
Are you sure they are stored in the cloud? I spoke to Skype support before making this script (to have them send me the chat history as required by the GDPR) and they claimed they only store it for 30 days in the cloud.

The Skype client has a bug where it doesn't allow you to scroll up without an Internet connection, because it loads something from the Internet while doing that (e.g. images), giving the impression that the whole conversation is stored in the cloud while in reality it is loaded locally.

The conversations for previous Skype clients are stored in the main.db file, which is a SQLite database. The new Skype client also has a feature to export those.

GIven the nature of Electron it is baslically all open source, so it should be quite easy to track down where it loads the conversations from the cloud, maybe you can even see it in the network tab of the debugging tools.

**Comment by Seblink** [comment]
> Are you sure they are stored in the cloud? I spoke to Skype support before making this script (to have them send me the chat history as required by the GDPR) and they claimed they only store it for 30 days in the cloud.

Yes, they are stored in the cloud. I am experimenting with a fresh os installation with the new Skype 8 on it, and I am able to go back until April 2017 at the moment for some conversations, but the big conversations causes a crash before it gets there.

I was also thinking about going the GDPR route, but you say they won't provide you with information older than 30 days? How did you contact them, did you just sent them an email?

> GIven the nature of Electron it is baslically all open source, so it should be quite easy to track down where it loads the conversations from the cloud, maybe you can even see it in the network tab of the debugging tools.

I am currently looking into that, but I'm not very familiar with Electron. Thanks for the advice.

**Comment by siilike** [comment]
They will have to provide you with everything they have stored, but they say they only store the conversations for 30 days. I used the Contact Us page: https://support.microsoft.com/en-gb/contactus/skype/ but they were just playing stupid like they didn't understand what I wanted, so it was easier and faster to just write the script.

If you are sure the conversations get loaded from the cloud, you should be able to see the relevant request in the network tab.

**Comment by sigmap** [comment]
I did a comparison of an export vs the gdpr route and I was able to find all message in both messages.json and export.json afaik without needing both. Both follow a different structure however and export,json doesn't come with media to reference.
